### DISCUSSION TITLE
Poor documentation for testing

### DISCUSSION THREAD

**Discussion by vahidvdn** [remix-run/remix#8517] [Q&A] [answered]
It's been a day that I'm stuck with implementing testing for components. The only document is the following link: https://remix.run/docs/en/main/other-api/testing

Which has no installation setup. I also checked [examples](https://github.com/remix-run/examples) repository, but could find any test there. Any idea would highly appreciated!

**Comment by kiliman** [comment, marked as answer]
I show how to setup testing using Vitest and Playwright in my `remix-vite-template`

https://github.com/kiliman/remix-vite-template/blob/main/docs/testing.md

**Reply by vahidvdn** [reply]
@kiliman Thanks. For e2e I prefer Cypress over Playwright

**Comment by jrestall** [comment]
https://github.com/epicweb-dev/full-stack-testing/tree/main/exercises/07.remix-component

**Reply by vahidvdn** [reply]
Thanks!
